### PR TITLE
Add grouping to the contest

### PR DESCRIPTION
**Pull request by belev**

### CODE REVIEW

File: app/policies/contest_group_policy.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  def destroy?
+    admin_or_coach?
+  end
+
+  private
+
+  def admin_or_coach?
+    user && (user.admin? || user.coach?)
+  end
+end
```
Comment (valo) [inline app/policies/contest_group_policy.rb]: Please, configure your editor to add a new line at the end of the files. This is a convention followed in unix: http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: app/views/admin/contest_groups/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+<br />
+<%= form_for [:admin, @contest_group] do |f| -%>
+    <p>
+      <%= f.error_messages %>
+    </p>
+    <%= render 'contest_group_fields', f: f %>
+    <br />
+    <p>
+      <%= f.submit "Обновяване" %><%= link_to "Отказ", admin_contest_groups_path %>
+    </p>
```
Comment (valo) [inline app/views/admin/contest_groups/edit.html.erb]: Please use 2 spaces for indentation. This is the convention that I try to use in the project.

File: app/views/admin/contest_groups/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<h1>Групи</h1>
+<br />
+
+<% if !@contest_groups.blank? %>
```
Comment (valo) [inline app/views/admin/contest_groups/index.html.erb]: Generally using `!` in the conditionals may lead to confusion sometimes. Rails provides things like `.present?` to avoid using `!`. It is not that important, so change it only if you want to :)

File: app/views/layouts/main.html.erb [hunk 4]
```diff
@@ -52,7 +52,7 @@
 </div>
 <!-- wrap END -->
 
-<%= include_gon %>
+<%= include_gon(init: true) %>
```
Comment (valo) [inline app/views/layouts/main.html.erb:55]: Why is this change needed?

File: app/views/main/_practice_area.html.erb [hunk 5]
```diff
@@ -2,27 +2,19 @@
...
+      <% @contest_groups.each do |contest_group| %>
+          <tr>
+            <td><%= link_to contest_group.name, contest_group_path(contest_group) %></td>
+          </tr>
+      <% end %>
+    </table>
+    <%= will_paginate @contest_groups, :param_name => :contest_groups_page %>
 <% else %>
-В момента няма състезания, на които да можете да се упражнявате.
+    В момента няма състезания, на които да можете да се упражнявате.
```
Comment (valo) [inline app/views/main/_practice_area.html.erb]: Please, keep the old list view of the contests. We don't want to confuse people with a completely new view. The group view and the list view can live on the main page for a while and then we can decide which one to keep.

File: app/views/main/problems.html.erb [hunk 6]
```diff
@@ -6,7 +6,7 @@
     <th>Име</th>
     <th>Състезание</th>
     <th>Пратени решения</th>
-    <th>Среден брой точки</p>
+    <th>Среден брой точки</th>
```
Comment (valo) [inline app/views/main/problems.html.erb]: Wow! Good catch!

File: app/views/main/index.html.erb [hunk 7]
```diff
@@ -25,7 +25,7 @@
   <%= render :partial => "practice_area" %>
 <% end %>
 <br />
-<h2>Състезания на други сайтове</h3>
+<h2>Състезания на други сайтове</h2>
```
Comment (valo) [inline app/views/main/index.html.erb:28]: Wow! Good catch!

File: features/admin/admin_login.feature [hunk 8]
```diff
@@ -1,7 +1,7 @@
 @admin
 Feature: Administrator login
   In order to be able to control the system
-  As a administrator
+  As an administrator
```
Comment (valo) [inline features/admin/admin_login.feature:4]: Nice :)

File: spec/controllers/admin/contests_controller_spec.rb [hunk 9]
```diff
@@ -40,7 +40,7 @@
     end
 
     describe "#create" do
-      let(:contest_params) { attributes_for(:contest) }
+      let(:contest_params) { build(:contest).attributes }
```
Comment (valo) [inline spec/controllers/admin/contests_controller_spec.rb:43]: I didn't know you can do this trick to get a hash with the IDs of the relations. Nice one!

File: spec/policies/contest_group_policy_spec.rb [hunk 10]
```diff
@@ -0,0 +1,28 @@
+describe ContestGroupPolicy do
+  let(:contest_group) { create(:contest_group) }
+  let(:admin) { create(:user, role: User::ADMIN) }
+  let(:coach) { create(:user, role: User::COACH) }
+  let(:contester) { create(:user, role: User::CONTESTER) }
+
+  [
+      :index?, :new?, :create?, :edit?, :update?, :destroy?
```
Comment (valo) [inline spec/policies/contest_group_policy_spec.rb]: Indentation issue here. Please, use 2 spaces.

File: app/policies/contest_group_policy.rb (hunk 1 above)
Comment (belev) [inline app/policies/contest_group_policy.rb]: Should new lines be added to .erb files too ?

File: app/views/layouts/main.html.erb (hunk 4 above)
Comment (belev) [inline app/views/layouts/main.html.erb:55]: I added it because I got undefined error in the browser console. After this the problem was gone.

File: app/views/main/_practice_area.html.erb (hunk 5 above)
Comment (belev) [inline app/views/main/_practice_area.html.erb]: So, just to clarify that I understand correctly, we want on the main page both views. The view with the contests and the new view, which will be showing the groups, is that right ?

File: app/policies/contest_group_policy.rb (hunk 1 above)
Comment (valo) [inline app/policies/contest_group_policy.rb]: Yes

File: app/views/layouts/main.html.erb (hunk 4 above)
Comment (valo) [inline app/views/layouts/main.html.erb:55]: Nice! Didn't know about this option. Much better than checking everywhere if gon is initialized
